src/lwc/healthCheckGroup/healthCheckGroup.js

```diff
@@ -0,0 +1,182 @@
+
+import getHealthCheckGroupViewModel from '@salesforce/apex/HealthCheckGroupController.getHealthCheckGroupViewModel';
+
+export default class HealthCheckGroup extends LightningElement {
+    @api isDisplayGroup;
+    @api healthCheckDefinition;
+
+    @track healthCheckItemList;
+    @track expandedRowsList;
+    @track totalChecks = 0;
+    @track passedChecks = 0;
+    @track isExpanded;
+    
+    labelReference = {
+        stgHealthChecksAllPassed,
+        stgHealthChecksNotAllPassed,
+        stgHealthCheckStatus,
+        stgHealthCheckSetting,
+        stgHealthCheckResultsDescription,
+        stgHealthCheckRecoFix,
+        stgHealthCheckRunError,
+        stgHealthCheckResultsExpand,
+        stgHealthCheckResultsCollapse
+    }
+
+    iconReference = {
+        expandedIcon: 'utility:chevrondown',
+        collapsedIcon: 'utility:chevronright',
+        errorIcon: 'utility:error',
+        successIcon: 'utility:success',
+    }
+
+    get healthCheckResultsId(){
+        let tempString = this.healthCheckDefinition.name + 'Results';
+        return tempString.split(" ").join("");
+    }
+
+    get collapsableIcon() {
+        if (this.isExpanded){
+            return this.iconReference.expandedIcon;
+        }
+
+        return this.iconReference.collapsedIcon;        
+    }
+
+    get collapsableIconAltText(){
+        if (this.isExpanded){
+            return this.labelReference.stgHealthCheckResultsCollapse;
+        }
+        
+        return this.labelReference.stgHealthCheckResultsExpand;
+    }
+
+    get statusIcon() {
+        if (this.isAllSuccessStatus()){
+            return this.iconReference.successIcon;
+        }
+
+        return this.iconReference.errorIcon;        
+    } 
+
+    get statusIconVariant() {
+        if (this.isAllSuccessStatus()) {
+            return 'success';
+        }
+
+        return 'error';
+    }
+
+    get passedChecksDisplay() {
+        if (this.passedChecks == this.totalChecks){
+            return this.labelReference.stgHealthChecksAllPassed;
+        }
+
+        return this.labelReference.stgHealthChecksNotAllPassed.replace(
+            '{0}', this.passedChecks
+        ).replace(
+            '{1}',this.totalChecks
+        );
+    }
+
+    isAllSuccessStatus(){
+        if (this.passedChecks == this.totalChecks){
+            return true;
+        }
+
+        return false;
+    }
+
+    toggleExpanded() {
+        this.isExpanded = !this.isExpanded; 
+    }
+
+    @wire(
+        getHealthCheckGroupViewModel, 
+        {
+            name: '$healthCheckDefinition.name', 
+            className: '$healthCheckDefinition.className', 
+            namespace: '$healthCheckDefinition.namespace'
+        })
+    treeData({ error, data }) {
+        console.log('wiring data');
+        if ( data ) {
+            let stringified = JSON.stringify(data);            
+            let tempData = JSON.parse( stringified );
```

Comment:
Let's talk about naming to indicate what we're doing here a bit more clearly. This is essentially a hack to get around wired object child array assignment issues.

Reply:
maybe a brief comment to describe the hack?

Reply:
We could call it arrayAssignmentDataHack. Whichever way you want to play it. I think this is one of those situations where names and a comment are warranted. The purpose of this is to fix an issue where a single-member array evaluates only to the member of that array. This prevents appropriate copying of the array to a differently-named array to support lightning tree grid child format.